src/test/java/fi/otavanopisto/kuntaapi/test/AbstractPtvMocker.java

```diff
@@ -0,0 +1,493 @@
+  }
+  
+  private MappingBuilder createOkGetMapping(UrlPattern urlPattern, R resource) {
+    return get(urlPattern)
+      .willReturn(createOkGetReturn(resource));
+  }
+  
+  private MappingBuilder createNotFoundMapping(UrlPattern urlPattern) {
+    return get(urlPattern)
+      .willReturn(aResponse()
+      .withStatus(404));
+  }
+  
+  private ResponseDefinitionBuilder createOkHeadReturn(R resource) {
+    if (resource instanceof File) {
+      File file = (File) resource;
+      
+      try (FileInputStream fileInputStream = new FileInputStream(file)) {
+        String contentType = Files.probeContentType(file.toPath());
+        byte[] data = IOUtils.toByteArray(fileInputStream);
+        return aResponse()
+          .withHeader(CONTENT_TYPE, contentType)
+          .withHeader(CONTENT_LENGTH, String.valueOf(data.length));
+      } catch (IOException e) {
+        fail(e.getMessage());
+        return null;
+      }
+      
+    } else {
+      return aResponse()
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON);
+    }
+  }
+  
+  private ResponseDefinitionBuilder createOkGetReturn(R resource) {
+    if (resource instanceof File) {
+      File file = (File) resource;
+      
+      try (FileInputStream fileInputStream = new FileInputStream(file)) {
+        String contentType = Files.probeContentType(file.toPath());
+        byte[] data = IOUtils.toByteArray(fileInputStream);
+        return aResponse()
+          .withHeader(CONTENT_TYPE, contentType)
+          .withHeader(CONTENT_LENGTH, String.valueOf(data.length))
+          .withBody(data);
+      } catch (IOException e) {
+        fail(e.getMessage());
+        return null;
+      }
+      
+    } else {
+      return aResponse()
+        .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+        .withBody(toJSON(resource));
+    }
+  }
+  
+  private void updateStatusLists() {
+    for (Entry<MockedResourceStatus, List<MappingBuilder>> statusListEntry : statusLists.entrySet()) {
+      List<MappingBuilder> mappings = statusListEntry.getValue();  
+      for (MappingBuilder mapping : mappings) {
+        MockedResourceStatus status = statusListEntry.getKey();
+        
+        removeStub(mapping);
+        
+        mapping.willReturn(aResponse()
+          .withHeader(CONTENT_TYPE, APPLICATION_JSON)
+          .withBody(toJSON(getGuidPage(status))));
+        
+        stubFor(mapping);
+      }
+    }
+  }
+  
+  private String toJSON(Object object) {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      objectMapper.registerModule(new JavaTimeModule());
+      return objectMapper.writeValueAsString(object);
+    } catch (JsonProcessingException e) {
+      fail(e.getMessage());
+      return null;
+    }
+  }
+  
+  public class GuidPage {
+
+    private Integer pageNumber = null;
+    private Integer pageSize = null;
+    private Integer pageCount = null;
+    private List<VmOpenApiItem> itemList = new ArrayList<VmOpenApiItem>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)